arch/arm/core/aarch32/mpu/arm_core_mpu.c

```diff
@@ -46,7 +46,7 @@ LOG_MODULE_REGISTER(mpu);
 		_MPU_DYNAMIC_REGIONS_AREA_START)
 
 #if !defined(CONFIG_MULTITHREADING) && defined(CONFIG_MPU_STACK_GUARD)
-K_THREAD_STACK_DECLARE(z_main_stack, CONFIG_MAIN_STACK_SIZE);
+extern K_THREAD_STACK_DEFINE(z_main_stack, CONFIG_MAIN_STACK_SIZE + CONFIG_COVERAGE_STACK_SIZE);
```

Comment:
This is incorrect. `K_THREAD_STACK_DECLARE` (aka. extern) must be used here.

Reply:
Thanks for correcting, I'll do

Reply:
Hi sorry for the delay, I've addressed it.